src/plugins/layout/states/style.ts

```diff
@@ -31,3 +32,17 @@ export function getCssRules(element: HTMLStyleElement): string[] {
 
     return rules;
 }
+
+export function shouldCaptureCssRules(node: Node): boolean {
+    let captureCssRules = false;
+    if (node.nodeType === Node.ELEMENT_NODE && (node as Element).tagName === "STYLE") {
+        // If 'cssRules' is set to true, capture rules on all style nodes.
+        // Otherwise, optimistically hope that rules match the textContent,
+        // except for the case when there is no inner text. 'Empty' <style>
+        // node is a rather common way to add styles to the page by using
+        // <style>.insertRule API instead of appending css text children.
```

Comment:
I couldn't quite follow this helpful comment. Can you clarify it? Might be faster to chat about quickly.

Reply:
Well, that means i should definitely rephrase it :) But for your clarity

a. If config.cssRules == true, for any style node, capture it's sheet rules and ignore text child nodes
b. If config.cssRules == false, only "capture rules and ignore text children" for style nodes that have no children. If there is a style node with no children, it's very very likely that page does some direct css rule modifications through <style>.insertRule using this node, so if we don't capture it, the page will most likely be broken.

Does that make more sense?

---
In reply to: [261422191](https://github.com/Microsoft/clarity-js/pull/138#discussion_r261422191) [](ancestors = 261422191)

Reply:
Updated comment

---
In reply to: [261423297](https://github.com/Microsoft/clarity-js/pull/138#discussion_r261423297) [](ancestors = 261423297,261422191)